projects/frontegg-app/src/lib/frontegg-app.service.ts

```diff
@@ -63,6 +64,32 @@ export class FronteggAppService {
     return this.isLoadingSubject.asObservable();
   };
 
+  /**
+   * @param keys Entitlements keys
+   * @param observer For receiving the keys data results
+   * @returns a subscription to be able to unsubscribe
+   */
+  public entitlements$(keys: string[], observer: PartialObserver<Entitlements>): Subscription {
+    // used for computing the entitlements result because we don't return the state itself
+    const entitlementsSubject = new BehaviorSubject<Entitlements>([]);
+    
+    const stateSubscription = this.entitlementsStateSubject.subscribe(entitlements => {
+      entitlementsSubject.next(getEntitlements(entitlements ?? {}, keys));
+    });
+    
+    // subscribing the consumer observer
+    const entitlementsResultSubscription = entitlementsSubject.asObservable().subscribe(observer)
```

Comment:
Why not saving entitlementsResultSubscription as part of the class? and then could be access from multiple places?

Reply:
cause we use different keys for every func call